packages/viz/src/lib/basemap.ts

```diff
@@ -0,0 +1,112 @@
+
+/**
+ * Mapping libraries are meant to be available in the browser before using these helpers
+ */
+declare global {
+  interface Window {
+    deck: {
+      DeckGL(params?: any): any;
+      GoogleMapsOverlay(params?: any): any;
+    };
+
+    google: {
+      maps: {
+        Map(mapContainer?: any, opts?: any): any;
+      }
+    };
+  }
+}
+
+interface StyleUrlCatalog { [key: string]: string; }
+const cartoBasemaps: StyleUrlCatalog = {
+  positron: 'https://basemaps.cartocdn.com/gl/positron-gl-style/style.json',
+  voyager: 'https://basemaps.cartocdn.com/gl/voyager-gl-style/style.json',
+  darkmatter: 'https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json',
+};
+
+/**
+ * A helper function to create a CARTO basemap on a 'map' DOM element, rendered using *Mapbox GL JS*
+ *
+ * Examples:
+ * ```javascript
+ *    // Several options to create the map are allowed
+ *    const deckMap = carto.viz.map();
+ *    const deckMap = carto.viz.map('voyager');
+ *    const deckMap = carto.viz.map('darkmatter', { zoom: 4 });
+ *    const deckMap = carto.viz.map('positron', { zoom: 4, longitude: 3, latitude: 40, pitch: 45, bearing: 30 }, 'map');
+ * ```
+ * @export
+ * @param {string} [basemap='positron']
+ * @param {*} [view]
+ * @param {string} [containerId='map']
+ * @returns
+ */
+export function map(basemap: string = 'positron', view?: any, containerId: string = 'map') {
```

Comment:
I think that instantiation like `carto.viz.map` is weird because you might think that you need to add `new` to instantiate a new map like Deck.gl or any other map libraries.

So, as it is not a class, why don't we create a `createMap` function?

When it comes to parameter definition, I think we should admit an object with the properties, because there's no difference in importance to have a main parameter and an options object.

So I think we can make something like this:
```js
const DEFAULT_VIEW = {
  longitude: 0,
  latitude: 0,
  zoom: 1
};

interface MapOptions {
  basemap: string;
  view: {
    longitude: number;
    latitude: number;
    zoom: number;
  },
  container: string;
};

const defaultOptions = {
  container: '#map',
  basemap: 'voyager',
  view: DEFAULT_VIEW
};

export function createMap(options: MapOptions = defaultOptions) {
}
```

Reply:
Yes, the idea was to simplify the function, but it's true it might be confusing, regarding to the 'missing' new. I will give it another try, changing name and params options...

Reply:
First step: clearer names --> `createMap`, `createGoogleMap` (both will come inside `carto.viz`)